test/DI.Tests/CircularDependencyTests.cs

```diff
@@ -2,8 +2,11 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Generic;
+using System.Linq;
 using Microsoft.Extensions.DependencyInjection.Tests.Fakes;
 using Xunit;
+using static Microsoft.Extensions.Internal.TypeNameHelper;
```

Comment:
I would recommend not using any helper here at all and putting the stringified names in the tests, which I think will make the tests easier to understand and verify because you can see what the exception messages will really look like.